### ISSUE TITLE
Block Comments Accessibility: block comment sidebar is expanded, but the screen reader doesn't announce it

### ISSUE THREAD

**Issue by jeffpaul** [WordPress/gutenberg#71675]
### Description

From @t-hamano in https://github.com/WordPress/gutenberg/issues/71249#issue-3333938703:

> Click the Comment button in the block toolbar:
> The block comment sidebar is expanded, but the screen reader doesn't announce it.

From @joedolson in https://github.com/WordPress/gutenberg/issues/71249#issuecomment-3201284909:

> Choosing the block option to add a comment should be considered as initiating an action, and focus should move to the comment box. I would suggest it should move directly to the comment input, as the clear intended action is to write a comment.

> Moving focus would also trigger an announcement, as the new focus location would be announced.

From @adamsilverstein in https://github.com/WordPress/gutenberg/issues/71249#issuecomment-3229794396:

> Choosing the block option to add a comment (or clicking the Comment button in the block toolbar) should move focus directly to the comment input field. (Moving focus would also trigger an announcement, as the new focus location would be announced).

### Step-by-step reproduction instructions

.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
